meilisearch/src/routes/indexes/documents.rs

```
@@ -574,6 +575,66 @@ pub async fn delete_documents_by_filter(
+    #[deserr(default, error = DeserrJsonError<InvalidDocumentFilter>)]
+    filter: Option<Value>,
+    #[deserr(default, error = DeserrJsonError<InvalidDocumentFilter>)]
+    context: Option<Value>,
+    #[deserr(error = DeserrJsonError<InvalidDocumentFilter>, missing_field_error = DeserrJsonError::missing_document_filter)]
+    function: String,
+}
+
+pub async fn edit_documents_by_function(
+    index_scheduler: GuardedData<ActionPolicy<{ actions::DOCUMENTS_ALL }>, Data<IndexScheduler>>,
+    index_uid: web::Path<String>,
+    body: AwebJson<DocumentEditionByFunction, DeserrJsonError>,
+    req: HttpRequest,
+    opt: web::Data<Opt>,
+    _analytics: web::Data<dyn Analytics>,
+) -> Result<HttpResponse, ResponseError> {
+    debug!(parameters = ?body, "Edit documents by function");
+    let index_uid = IndexUid::try_from(index_uid.into_inner())?;
+    let index_uid = index_uid.into_inner();
+    let DocumentEditionByFunction { filter, context, function } = body.into_inner();
+
+    // analytics.delete_documents(DocumentDeletionKind::PerFilter, &req);
+
+    let engine = milli::rhai::Engine::new();
+    if let Err(e) = engine.compile(&function) {
+        return Err(ResponseError::from_msg(e.to_string(), Code::BadRequest));
+    }
+
+    if let Some(ref filter) = filter {
+        // we ensure the filter is well formed before enqueuing it
+        || -> Result<_, ResponseError> {
+            Ok(crate::search::parse_filter(filter)?.ok_or(MeilisearchHttpError::EmptyFilter)?)
+        }()
+        // and whatever was the error, the error code should always be an InvalidDocumentFilter
+        .map_err(|err| ResponseError::from_msg(err.message, Code::InvalidDocumentFilter))?;
+    }
+    let task = KindWithContent::DocumentEdition {
+        index_uid,
+        filter_expr: filter,
+        context: context.map(|v| match v {
+            serde_json::Value::Object(m) => m,
+            _ => panic!("The context must be an Object"),
```

I need to return a real error here.